storage/replica_trigger.go

```diff
@@ -320,26 +320,38 @@ func (r *Replica) leasePostCommitTrigger(
 					r.gossipFirstRangeLocked(ctx)
 				}()
 			}
-		} else if trigger.lease.Covers(r.store.Clock().Now()) {
-			if err := r.withRaftGroup(func(raftGroup *raft.RawNode) error {
-				if raftGroup.Status().RaftState == raft.StateLeader {
-					// If this replica is the raft leader but it is not the new lease
-					// holder, then try to transfer the raft leadership to match the
-					// lease.
-					log.Infof(ctx, "range %v: replicaID %v transfer raft leadership to replicaID %v",
-						r.RangeID, replicaID, trigger.lease.Replica.ReplicaID)
```

Comment:
Yeah, that's the one I was referring to.

On Sep 5, 2016 09:55, "Peter Mattis" <notifications@github.com> wrote:

> In storage/replica_trigger.go
> <https://github.com/cockroachdb/cockroach/pull/9102#discussion_r77525528>:
>
> > @@ -320,26 +320,38 @@ func (r *Replica) leasePostCommitTrigger(
> >  					r.gossipFirstRangeLocked(ctx)
> >  				}()
> >  			}
> > -		} else if trigger.lease.Covers(r.store.Clock().Now()) {
> > -			if err := r.withRaftGroup(func(raftGroup *raft.RawNode) error {
> > -				if raftGroup.Status().RaftState == raft.StateLeader {
> > -					// If this replica is the raft leader but it is not the new lease
> > -					// holder, then try to transfer the raft leadership to match the
> > -					// lease.
> > -					log.Infof(ctx, "range %v: replicaID %v transfer raft leadership to replicaID %v",
> > -						r.RangeID, replicaID, trigger.lease.Replica.ReplicaID)
>
> I removed the TODO(tschottdorf) I added in this PR. Not sure if you're
> referring to another one. Your comments don't seem to be anchored to the
> code correctly.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/9102/files/9426b8ffc228132a577ac5513684b279aebe1eea#r77525528>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPDf3QdgTD9AZwWqUu3Ov7cA-k603ks5qnB9fgaJpZM4J0qfB>
> .
>